WordPress/gutenberg
# StyleProvider: Convert component to TypeScript

## What?
Converts the `StyleProvider` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `StyleProvider` to TypeScript.

## Testing Instructions
1. Run `npm run dev` and ensure no typing errors
2. Test the Gutenberg editor that everything works as before.
